src/app/portlets/dot-edit-page/template-containers-cache.service.ts

```diff
@@ -9,7 +9,15 @@ export class TemplateContainersCacheService {
     private containers: { [key: string]: { container: DotContainer } };
 
     set(containers: { [key: string]: { container: DotContainer } }): void {
-        this.containers = containers;
+        const mappedContainers: { [key: string]: { container: DotContainer } } = {};
+        Object.keys(containers).forEach(function(item: string) {
+            if (containers[item].container.source === 'FILE') {
+                mappedContainers[containers[item].container.path] = containers[item];
+            } else {
+                mappedContainers[item] = containers[item];
+            }
+        });
+        this.containers = mappedContainers;
```

Comment:
why not assing directly to this.containers?
if we create the getContainerId method that maybe I said before we can use it here too

```
mappedContainers[this.getContainerId(containers[item])] = containers[item];
```

Reply:
Object.keys creates a new scope, `this.containers` is private